### ISSUE TITLE
Problem about validation and inference

### ISSUE THREAD

**Issue by mangoyuan** [MIC-DKFZ/nnUNet#57]
Thanks for your sharing. It is great!
My problem is that 
1) when training finishs, the run_training.py will automatically use the last model to validate and output the summary.json. 
   But in predict_simple.py, it will select best model to predict. Why the model selected during validation and inference phase is different? It's there any experiences about selecting models?
   Though we can use the --valbest in training, but the help of this args is "hands off. This is not intended to be used". 
Thanks for reading.

**Comment by FabianIsensee** [comment]
Hi,
nnunet selects the best epoch based on the validation metric (green line). If you validate with the best epoch then you are effectively overfitting to the validation set and will get an optimistic result for your cross-validation. It is cleaner to evaluate with the latest epoch. 
Then in inference you are evaluating on the test set. Here we can do with the training data as we please so it's ok to use the best epoch
Best,
Fabian

**Comment by mangoyuan** [comment]
Sincerely thanks for your fast reply. It really helps me a lot. :-)
